x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/ShardChangesAction.java

```diff
@@ -397,7 +398,11 @@ protected void doExecute(Task task, Request request, ActionListener<Response> li
                 if (cause instanceof IllegalStateException && cause.getMessage().contains("Not all operations between from_seqno [")) {
                     String message = "Operations are no longer available for replicating. Maybe increase the retention setting [" +
                         IndexSettings.INDEX_SOFT_DELETES_RETENTION_OPERATIONS_SETTING.getKey() + "]?";
-                    listener.onFailure(new ElasticsearchException(message, e));
+                    // Make it easy to detect this error in ShardFollowNodeTask:
```

Comment:
Maybe introduce a new exception (extends from IllegalStateException) which indicates that the requesting history is no longer available. We can throw that exception in the LuceneChangesSnapshot. WDYT?

Reply:
Thanks @martijnvg :)